kiro-synth-host/src/midi/drivers/jackmidi.rs

```diff
@@ -0,0 +1,80 @@
+use anyhow::Result;
+use thiserror::Error;
+
+use kiro_midi_core::decoder::{CallbackResult, Decoder, DecoderCallbacks};
+use kiro_midi_core::messages::Message;
+
+use crate::midi::drivers::{MidiError, MidiHandler};
+
+use jack::{Client, Control, ProcessHandler, ProcessScope};
+
+struct MidiDecoderCallbacks<'a, Handler: MidiHandler> {
+  timestamp: u64,
+  handler: &'a mut Handler,
+}
+
+impl<'a, Handler> DecoderCallbacks for MidiDecoderCallbacks<'a, Handler>
+where
+  Handler: MidiHandler,
+{
+  fn on_message(&mut self, message: Message) -> CallbackResult {
+    self.handler.on_message(self.timestamp, message);
+    CallbackResult::Continue
+  }
+
+  fn on_sysex(&mut self, data: &[u8]) -> CallbackResult {
+    self.handler.on_sysex(self.timestamp, data);
+    CallbackResult::Continue
+  }
+}
```

Comment:
I'm wondering if you could extract this up to `mod.rs` so it can be re-used between coremidi and jackmidi.